collection.go

```diff
@@ -107,6 +107,8 @@ func (cs *CollectionSpec) RewriteMaps(maps map[string]*Map) error {
 	return nil
 }
 
+var ErrMissingConstants = errors.New("spec is missing one or more constants")
```

Comment:
Judging from the test this should really be a dedicated error:

```go
type MissingConstantsError struct{ Constants []string }
```

Then the test can `errors.As` and assert on `err.Constants` instead of string comparison. WDYT?

Please update the documentation on RewriteConstants to indicate that a specific error is returned.